src/biome/text/api_new/pipeline.py

```diff
@@ -636,40 +632,82 @@ async def status():
         uvicorn.run(make_app(), host="0.0.0.0", port=port)
 
     @classmethod
-    def _to_allennlp_configuration(cls, config: TrainConfiguration) -> Dict[str, Any]:
-        trainer = {
-            **yaml_to_dict(config.trainer),
-        }
-        trainer["trainer"]["type"] = "default"
-        # Add cuda device if necessary
-        trainer["trainer"]["cuda_device"] = trainer["trainer"].get(
-            "cuda_device", get_env_cuda_device()
-        )
+    def _allennlp_configuration(
+        cls, pipeline: Pipeline, config: _TrainConfiguration
+    ) -> Dict[str, Any]:
+        def iterator_configuration() -> Dict[str, Any]:
+            """Creates a data iterator configuration"""
+
+            def _forward_inputs(pipeline: Pipeline) -> List[str]:
+                """
+                Calculate the required head.forward arguments. We use this method
+                for automatically generate data iterator sorting keys
+                """
+                required, _ = split_signature_params_by_predicate(
+                    pipeline.head.forward,
+                    lambda p: p.default == inspect.Parameter.empty,
+                )
+                return [p.name for p in required] or [None]
+
+            iterator_config = {
+                "batch_size": config.trainer.batch_size,
+                "max_instances_in_memory": config.trainer.batch_size * 3,
```

Comment:
Maybe i would also expose the magic number 3 to the config.trainer with the default value 3.
During the DZ project i used an value of `batch_size * 1` for example.